qos/evm/response.go

```diff
@@ -0,0 +1,121 @@
+package evm
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// responseUnmarshaller is the entrypoint function for any
+// new supported response types, e.g. to add a response to an
+// eth_getBalance request, a new responseUnmarshaller needs to
+// be defined, along with a custom struct to handle the details
+// of the particular response.
+type responseUnmarshaller func([]byte) (response, error)
+
+var (
+	_ response = &responseToChainID{}
+	_ response = &responseToBlockHeight{}
+
+	methodResponseMappings = map[method]responseUnmarshaller{
+		methodChainID:     responseUnmarshallerChainID,
+		methodBlockNumber: responseUnmarshallerBlockHeight,
+	}
+)
+
+func unmarshalResponse(method method, data []byte) (response, error) {
+	unmarshaller, found := methodResponseMappings[method]
+	if found {
+		return unmarshaller(data)
+	}
+
+	return genericUnmarshaller(data)
+}
+
+func responseUnmarshallerChainID(data []byte) (responseToChainID, error) {
+	var response responseToChainID
+	if err := json.Unmarshal(data, &response); err != nil {
+		return responseToChainID{}, err
+	}
+
+	return response, nil
+}
+
+// TODO_IN_THIS_COMMIT: implement this unmarshaller
+func responseUnmarshallerBlockHeight(data []byte) (responseToBlockHeight, error) {
+
+}
+
+// TODO_IN_THIS_COMMIT: implement this unmarshaller
+func responseUnmarshallerGeneric(data []byte) (responseGeneric, error) {
+
+}
+
+type responseToChainID struct {
```

Comment:
Thoughts on having a separate file for each of these?

I can be swayed both ways, but think it'll become easier to maintain over a longer period of time.